src/main.rs

```diff
@@ -90,14 +90,23 @@ fn print_page(
         let reader = BufReader::new(file);
 
         if enable_markdown {
+            let mut stdout = std::io::stdout();
+            stdout.lock();
             // Print the raw markdown of the file.
             for line in reader.lines() {
-                println!("{}", line.unwrap());
+                writeln!(stdout, "{}", line.unwrap())
+                    .map_err(|_| "Could not write to stdout".to_string())?;
             }
+            stdout
+                .flush()
+                .map_err(|_| "Could not flush stdout".to_string())?;
         } else {
             // Create tokenizer and print output
             let mut tokenizer = Tokenizer::new(reader);
-            print_lines(&mut tokenizer, &config);
+            let mut stdout = std::io::stdout();
+            stdout.lock();
+            print_lines(&mut stdout, &mut tokenizer, &config);
+            stdout.flush().map_err(|err| err.to_string())?;
```

Comment:
This can also be done after this conditional block, why exactly is a flush error reported differently depending on `enable_markdown`?

Reply:
Just didn’t cross my mind 🤷‍♂️